File: 프로그래머스/3/43238. 입국심사/입국심사.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

long long solution(int n, vector<int> times) {
    long long answer = 0;
    sort(times.begin(), times.end());
    
    long long left = 1;
    long long right = 1e14;
    
    while(left <= right){
        long long mid = (left+ right)/2;
        
        long long  tmp = 0;
        for(auto time : times){
            tmp+=mid/time;
        }
        
        if(tmp >= n){
            answer = mid;
            right = mid -1;
        }else{
            left = mid +1; 
        }
    }
    
    return answer;
}

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    var answer = 0;
    
    let left = 0;
    let right = 1e15;
    
    while(left<=right){
        let mid = Math.floor( (left + right)/2);
        
        let howMany = 0;
        
        times.forEach((t) =>{
            howMany += Math.floor(mid / t);
        });
        
        if(howMany >= n){
            answer = mid;
            right = mid - 1;
        }else{
            left = mid + 1;
        }
    }
    return answer;
}